File: ant-sdk/antd-ruby/examples/03_chunks.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Example 03: Chunk put/get

require_relative "../lib/antd"

client = Antd::Client.new

# Store a raw chunk
result = client.chunk_put("raw chunk data")
puts "Chunk stored at #{result.address} (cost: #{result.cost} atto)"

# Retrieve the chunk
data = client.chunk_get(result.address)
puts "Retrieved chunk: #{data}"

File: ant-sdk/antd-zig/examples/03-chunks.zig
const std = @import("std");
const antd = @import("antd");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();

    var client = antd.Client.init(allocator, antd.default_base_url);
    defer client.deinit();

    // Store a chunk
    const chunk_data = "raw chunk content";
    std.debug.print("Storing chunk...\n", .{});

    const put_result = try client.chunkPut(chunk_data);
    defer put_result.deinit(allocator);

    std.debug.print("Chunk stored at: {s}\n", .{put_result.address});
    std.debug.print("Cost: {s} atto\n", .{put_result.cost});

    // Retrieve the chunk
    const data = try client.chunkGet(put_result.address);
    defer allocator.free(data);

    std.debug.print("Retrieved chunk: {s}\n", .{data});
}
